jax/experimental/jax2tf/jax2tf.py

```diff
@@ -126,6 +127,119 @@ def add_unit(v: TfValOrUnit, aval: core.AbstractValue):
     return core.unit if aval is core.abstract_unit else v
   return util.safe_map(add_unit, vals, avals)
 
+
+class SingletonDecorator:
+  def __init__(self, cls):
+    self.cls = cls
+    self.instance = None
+
+  def __call__(self, *args, **kwargs):
+    if not self.instance:
+      self.instance = self.cls(*args, **kwargs)
+    return self.instance
+
+PrimitiveName = str
+ErrorType = str
+ErrorString = str
+Devices = Tuple[str, ...]
+Limitation = Tuple[PrimitiveName, ErrorType, ErrorString, Devices]
```

Comment:
Maybe it is better to use a collections.namedtuple here?

Reply:
Addressed!